### PR TITLE
Httpclient improvements

### PR DESCRIPTION
**Pull request by LemonBoy**
I really hate github.

### CODE REVIEW

**Comment by zy93xy** [comment]
Use it to invoke applications on other servers,It can be quite different from a browser in terms of call time.
example：
 Google browser takes about 800 milliseconds
image
httpclient  takes about 100 milliseconds
image

**Comment by narimiran** [comment]
@dom96 what do you think of these changes?

File: lib/pure/httpclient.nim [hunk 1]
```diff
@@ -930,26 +923,26 @@ proc requestAux(client: HttpClient | AsyncHttpClient, url: string,
...
 
-  let getBody = httpMethod.toLowerAscii() notin ["head", "connect"] and
+  let getBody = httpMethod notin ["HEAD", "CONNECT"] and
                 client.getBody
   result = await parseResponse(client, getBody)
 
-proc request*(client: HttpClient | AsyncHttpClient, url: string,
-              httpMethod: string, body = "",
+proc request*(client: HttpClient | AsyncHttpClient, url: Uri | string,
+              httpMethod: HttpMethod | string = "GET", body = "",
```
Comment (dom96) [inline lib/pure/httpclient.nim]: This should probably be HttpGet by default. Also, AFAIK @Araq dislikes this pattern so might be a good idea to ask him for a review.

**Comment by ringabout** [comment]
suspended by #15919

**Pull request closed without merging**